rosetta/transactions/parse_transaction.go

```diff
@@ -15,16 +15,208 @@
 package transactions
 
 import (
+	"bytes"
 	"fmt"
+	"strconv"
 
+	"github.com/onflow/cadence/encoding/json"
 	"github.com/onflow/flow-go-sdk"
 
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/failure"
 	"github.com/optakt/flow-dps/rosetta/identifier"
 	"github.com/optakt/flow-dps/rosetta/object"
 )
 
-// ParseTransactions processes the flow transaction and translates it to a list of operations and a list of
-// signers.
+const (
+	// Transactions should have exactly one authorizer.
+	authorizersRequired = 1
+
+	// Transaction script should have exactly two arguments.
+	argsRequired = 2
+)
+
+// ParseTransactions processes the flow transaction, validates its correctness and translates it
```

Comment:
This is now more of a 'parse-and-validate-transaction'. Originally I liked the idea of the function returning operations and signers after parsing. Now with the extensive validation, the function has grown quite a bit, ~180 lines of code.

Perhaps it could be split to two parts - parsing transaction (validate and get the operations), and extracting the signatures (and validating the number of signatures). That would make it smaller, but also feels a bit like decomposition for the sake of decomposition.

I don't mind it either way..

Reply:
I think if we refactor this, we will do so on the final feature PR. However, you bring up an interesting point. Do we need to validate the signature on a signed transaction? And do we?

Reply:
We don't. I will add some more validation code in the `Submit` endpoint, to make sure that we actually have signatures (and the signer is the one we need), before we even submit the transaction.

Is that what you hand in mind?

Or some **real** validation, like what's done in [Flow code](https://github.com/onflow/flow-go/blob/master/fvm/transactionVerifier.go) when processing a transaction